.github/patches/duckdb-r/left_semi_anti.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/src/relational.cpp b/src/relational.cpp
+index 15f8469a..889145c0 100644
+--- a/src/relational.cpp
++++ b/src/relational.cpp
+@@ -327,9 +327,9 @@ bool constant_expression_is_not_null(duckdb::expr_extptr_t expr) {
+ 	} else if (join == "outer") {
+ 		join_type = JoinType::OUTER;
+ 	} else if (join == "semi") {
+-		join_type = JoinType::SEMI;
++		join_type = JoinType::LEFT_SEMI;
```

Comment:
It seems like there are a lot of changes related to renaming `SEMI` to `LEFT_SEMI`, and `ANTI` to `LEFT_ANTI`, but I wonder if this is necessary/a good idea to begin with? Right and left semi join are *semantically* the same, with only a difference in the execution layer, yet many changes here are made at the logical plan layer where there is no difference. Maybe it would be cleaner to leave the JoinType alone entirely and instead introduce a new operator similar to how the `LogicalIndexJoin` works?